myst_nbs/time_series/Air_passengers-Prophet_with_Bayesian_workflow.myst.md

```diff
@@ -6,39 +6,46 @@ jupytext:
     format_version: 0.13
     jupytext_version: 1.13.7
 kernelspec:
-  display_name: pymc3-dev-py38
+  display_name: Python 3
   language: python
-  name: pymc3-dev-py38
+  name: python3
 ---
 
+(Air_passengers-Prophet_with_Bayesian_workflow)=
 # Air passengers - Prophet-like model
 
-We're going to look at the "air passengers" dataset, which tracks the monthly totals of a US airline passengers from 1949 to 1960. We could fit this using the [Prophet](https://facebook.github.io/prophet/) model (indeed, this dataset is one of the examples they provide in their documentation), but instead we'll make our own Prophet-like model in PyMC3. This will make it a lot easier to inspect the model's components and to do prior predictive checks (an integral component of the [Bayesian workflow](https://arxiv.org/abs/2011.01808)).
+:::{post} April, 2022
+:tags: time series, prophet
+:category: intermediate
+:author: Marco Gorelli
+:::
+
++++
+
+We're going to look at the "air passengers" dataset, which tracks the monthly totals of a US airline passengers from 1949 to 1960. We could fit this using the [Prophet](https://facebook.github.io/prophet/) model {cite:p}`taylor2018forecasting` (indeed, this dataset is one of the examples they provide in their documentation), but instead we'll make our own Prophet-like model in PyMC3. This will make it a lot easier to inspect the model's components and to do prior predictive checks (an integral component of the [Bayesian workflow](https://arxiv.org/abs/2011.01808) {cite:p}`gelman2020bayesian`).
 
 ```{code-cell} ipython3
 import arviz as az
 import matplotlib.dates as mdates
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
-import pymc3 as pm
-
-print(f"Running on PyMC3 v{pm.__version__}")
+import pymc as pm
 ```
 
 ```{code-cell} ipython3
-RANDOM_SEED = 4000
+RANDOM_SEED = 8927
 rng = np.random.default_rng(RANDOM_SEED)
-
-%config InlineBackend.figure_format = 'retina'
 az.style.use("arviz-darkgrid")
+%config InlineBackend.figure_format = 'retina'
 ```
 
 ```{code-cell} ipython3
+# get data
 try:
     df = pd.read_csv("../data/AirPassengers.csv", parse_dates=["Month"])
 except FileNotFoundError:
-    df = pm.get_data("AirPassengers.csv", parse_dates=["Month"])
+    df = pd.read_csv(pm.get_data("AirPassengers.csv"), parse_dates=["Month"])
 ```
 
 ```{code-cell} ipython3
```

Comment:
If you have arviz>=0.12.0 this `_sample` thing is no longer needed, you can use https://python.arviz.org/en/v0.12.0/api/generated/arviz.extract_dataset.html